### PR TITLE
feat: recommend imperative mood, with emphais for `fix`

### LINKED ISSUE #190: ambiguity around description field of "fix"
The descrption specification example of fix is much unlike the descrption given in the summary for `fix`, and indeed, for every single other type. Compare:

```
(Summary)
feat: allow provided config object to extend other configs
feat(lang): add polish language
fix: correct minor typos in code 

(Spec)
fix: array parsing issue when multiple spaces were contained in string.
```

See the problem? Although the spec asks for the description part to describe what the commit does, its very own fix example fails to do that. It thinks the `fix` type field is good enough for a verb to describe what it does, which it does not.

Devs blindly following that specific example would write awful commit messages like `fix: foo bar crashes / not responds / does not work` without describing what the commit does to achieve it. And that defeats the point of conventional commit.

### PR DESCRIPTION
**Pull request by Artoria2e5**
This closes #190. The first commit is recovered from #85, which got reverted somehow some time later for some reason.

### CODE REVIEW

**Comment by bcoe** [comment]
@Artoria2e5 bother you to update your PR to reflect the `1.0.0` version of the specification we just landed?

File: content/next/index.md [hunk 1]
```diff
@@ -96,7 +96,8 @@ The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL
 1. A scope MAY be provided after a type. A scope MUST consist of a noun describing a
   section of the codebase surrounded by parenthesis, e.g., `fix(parser):`
 1. A description MUST immediately follow the colon and space after the type/scope prefix.
-The description is a short summary of the code changes, e.g., _fix: array parsing issue when multiple spaces were contained in string._
+The description is a short summary of the code changes, e.g., _fix: don't ignore consecutive spaces when parsing array._
+1. When the type field is a verb, the description MAY be an object or action the verb is applied to, only when the change is trivial or clear enough from the type, e.g., _revert: drop Node 6 from testing matrix._ Nontrivial changes, especially those of a  `fix` type, SHOULD describe the changeset itself rather than the undesirable behavior it removes.
```
Comment (bcoe) [inline content/next/index.md]: I think I'm more comfortable with an FAQ suggestion, that speaks to some of these recommendations. It seems like we could keep the formal spec text centered mainly on logic that could be adopted by a parser; whereas this recommendation is more a guideline for how one might go about writing commit messages ...

It almost makes me wonder if we'd do better to add a "related reading" section, rather than continuing to expand the FAQ.

File: content/next/index.md [hunk 2]
```diff
@@ -187,6 +187,13 @@ revert: let us never again speak of the noodle incident
 Refs: 676104e, a215868
 ```
 
+### What writing form should I use?
+
+We recommend writing a commit description and body using the [imperative](https://en.wikipedia.org/wiki/Imperative_mood) present tense writing form. There are a significant number of examples of this writing form for commits: [1](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)[2](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#subject)[3](https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project)[4](https://medium.com/@danielfeelfine/commit-verbs-101-why-i-like-to-use-this-and-why-you-should-also-like-it-d3ed2689ef70)[5](https://chris.beams.io/posts/git-commit/#imperative)
+
+If you are using a verb, such as "update" and "revert" as a type, you may see it as the verb of the imperative phrase only if the change is really trivial. For more complex things such as "fix", you should stick to describing what exactly you have done using the imperative form. Writing things like _fix: crash on login_ is a huge no-no.
```
Comment (damianopetrungaro) [inline content/next/index.md]: Do not really agree with it `fix: crash on login`. Because it may have a description to explain more details.

Could you please elaborate more?

Comment (Artoria2e5) [reply content/next/index.md]: I can rephrase it to "Writing things like... without further description in the commit body is a huge no-no. You should at least try to mention the root cause."

The story is that I got fed up with someone else writing `fix: fooBarProcess jams up console` when `fix: turn off console logging for fooBarProcess` works much better. Crashes and more complex bugs are indeed harder to describe in 44 characters, although people should still try to write stuff like `fix: free up fooObj to avoid instant OOM death` in my opinion. The "huge" part really needs some re-considering.

Comment (damianopetrungaro) [reply content/next/index.md]: Any update on this @Artoria2e5 ?

**Comment by damianopetrungaro** [comment]
Any update @Artoria2e5 ?
Do you still have time to deliver this? So much value! We'd love to see you as contributor for this!

**Comment by wesleytodd** [comment]
I commented over in #190 and didn't realize there was a PR for this: https://github.com/conventional-commits/conventionalcommits.org/issues/190#issuecomment-701535204

> I am not sure that the specification can add much value in this area. I think that a guide/faq could provide recommendations, but as there is now way to validate/parse or otherwise tool around the meaning of the description I don't think there is value in the spec trying to detail expectations there.

File: content/next/index.md (hunk 2 above)
Comment (smlx) [reply content/next/index.md:194]: The imperative style is quite well established good practice. See for example https://www.git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project

**Comment by damianopetrungaro** [comment]
@Artoria2e5 any update? I'd love to see this one landing tho

File: content/next/index.md (hunk 2 above)
Comment (jcornaz) [reply content/next/index.md:194]: I do aggree with imperative style, but I don't aggree with this specific example.

The commit-message, especially in the context of conventional-commit, should be a user-oriented changelog. And a user shouldn't have to care nor understand the actual cause of the issue. We only tell the user what is fixed, not the internal details that caused the bug. Ofc, the message should contain under which conditions the bug was appearing, exaample: `fix: OOM when calling fooObj`

The body of the commit message may eventually contains some more in-depth/developer-oriented explanation of what was the root cause how it is fixed.

**Comment by tristan-cottam** [comment]
@Artoria2e5 is apparently "Functionally asleep", despite this issue remaining as relevant. How can this merge request be appropriately merged nonetheless?

**Comment by damianopetrungaro** [comment]
Going to close as nothing as been alive for a while.

If you feel this is still relevant, @TristanCottam / @Artoria2e5 feel free to open a new issue/PR to keep that moving!
